src/app/icd/ICDManager.cpp

```diff
@@ -15,12 +15,157 @@
  *    limitations under the License.
  */
 
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
 #include <app/icd/ICDManager.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/ConnectivityManager.h>
+#include <platform/LockTracker.h>
+#include <platform/internal/CHIPDeviceLayerInternal.h>
 
 namespace chip {
 namespace app {
 
-ICDManager::ICDManager() {}
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::IcdManagement;
 
+void ICDManager::ICDManager::Init()
+{
+    uint32_t activeModeInterval;
+    Attributes::ActiveModeInterval::Get(kRootEndpointId, &activeModeInterval);
+    VerifyOrDie(kFastPollingInterval.count() < activeModeInterval);
+    UpdateIcdMode();
+    UpdateOperationState(OperationalState::ActiveMode);
+}
+
+void ICDManager::ICDManager::Shutdown()
+{
+    // cancel any running timer of the icd
+    DeviceLayer::SystemLayer().CancelTimer(OnIdleModeDone, this);
+    DeviceLayer::SystemLayer().CancelTimer(OnActiveModeDone, this);
+    mIcdMode          = ICDMode::SIT;
+    mOperationalState = OperationalState::IdleMode;
+}
+
+bool ICDManager::SupportsCheckInProtocol()
+{
+    bool success;
+    uint32_t featureMap;
+    success = (Attributes::FeatureMap::Get(kRootEndpointId, &featureMap) == EMBER_ZCL_STATUS_SUCCESS);
+
+    return success ? ((featureMap & to_underlying(Feature::kCheckInProtocolSupport)) != 0) : false;
+}
+
+void ICDManager::UpdateIcdMode()
+{
+    assertChipStackLockedByCurrentThread();
+
+    ICDMode tempMode = ICDMode::SIT;
+
+    // The Check In Protocol Feature is required and the slow polling interval shall also be greater than 15 seconds
```

Comment:
This should presumably have a TODO comment pointing to an issue.

Reply:
There is a to-do. I'll move it up at the start of the comments. Do you want me to link the todo to a git issue?

Reply:
@jmartinez-silabs Yes, please link to issue here.